### DISCUSSION TITLE
How can you fake a login?

### DISCUSSION THREAD

**Discussion by AGeck113** [neuefische/questions#463] [Web] [answered]
Can someone explain, how we can fake a login?
It would be nice to know, with which data structure we could work and how it would look, so that we can prepare a little bit for that.

**Comment by JoergWolff** [comment]
Hi AGeck113,
how about a JSON file like `fakeUser=[{email: test@testmail.com, pswd: test}]`.
Then you can create a form with two input fields.
For control you can map over the JSON and search for the right entry.
Later on you can move it to a DB. But pswd should then be encrypted.

**Comment by doemser** [comment, marked as answer]
The best way to work with this challenge is to **assume that you're already logged in**. This will kill most cases where you want to fake a login and what I mean by that is the classic login form.

There are basically two possibilities:
1. you actually need a login and then you should already build it the right way.
2. you don't need a login and therefore shouldn't add one now.

Being logged in is in the end characterized by having a user object available, which could look like this:

```javascript
const user = {
  id: "unique-user-id",
  name: "Alex",
  email: "alex@alex.com",
  image: "https://www.an-image-url.com"
};

```
> :white_check_mark: You can mock this data just like the other data.

---

In your frontend you will then later only do something like this to show content only to users with a session:

```javascript
if(session){
    return <>...</>
}

```

> :x: Do not mock this. This is rather easy to implement later on.

---

If you want entries to be associated with your user, it will be necessary to attach an identifier to your entries.


```diff 

const recipes = [
  {
    id: "unique-recipe-id",
+  userId: "unique-user-id",
    name: "Hot Chocolate",
    ingredients: ["milk", "chocolate"]
  },
  {...}
];


```

---

To sum it up clearly: Do not build a fake login form, that really makes no sense for the agile process we are working with.

<img width="491" alt="Screenshot 2023-01-25 at 10 06 19" src="https://user-images.githubusercontent.com/93415777/214522906-3f597922-6f23-4e64-8029-fa857f82b951.png">
